packages/panna-sdk/src/core/utils/activity.ts

```diff
@@ -0,0 +1,454 @@
+ * // result: {
+ * //   activities: [{
+ * //     activityType: 'sent',
+ * //     transactionID: '0x...',
+ * //     amount: {
+ * //       type: 'eth',
+ * //       value: '100000000000',
+ * //       tokenInfo: {
+ * //         address: '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee',
+ * //         name: 'Ether',
+ * //         symbol: 'ETH',
+ * //         decimals: 18,
+ * //         type: 'eth'
+ * //       }
+ * //     },
+ * //     status: 'ok'
+ * //   }, {
+ * //     activityType: 'received',
+ * //     transactionID: '0x...',
+ * //     amount: {
+ * //       type: 'erc-20',
+ * //       value: '100000000000',
+ * //       tokenInfo: {
+ * //         address: '0x...',
+ * //         name: 'Lisk',
+ * //         symbol: 'LSK',
+ * //         decimals: 18,
+ * //         type: 'erc-20'
+ * //         icon: null
+ * //       }
+ * //     },
+ * //     status: 'ok'
+ * //   }, {
+ * //     activityType: 'received',
+ * //     transactionID: '0x...',
+ * //     amount: {
+ * //       type: 'erc-721',
+ * //       tokenId: '9500',
+ * //       instance: {
+ * //         id: '9500',
+ * //         isUnique: null,
+ * //         owner: null,
+ * //         tokenInfo: {
+ * //           address: '0x...',
+ * //           name: 'Lisk of Life',
+ * //           symbol: 'LOL',
+ * //           decimals: null,
+ * //           type: 'erc-721',
+ * //           icon: null
+ * //         }
+ * //       },
+ * //     status: 'ok'
+ * //   }, {
+ * //     activityType: 'minted',
+ * //     transactionID: '0x...',
+ * //     amount: {
+ * //       type: 'erc-1155',
+ * //       tokenId: '29900446396079726096...',
+ * //       value: '100000000000',
+ * //       tokenInfo: {
+ * //         address: '0x...',
+ * //         name: 'Rarible',
+ * //         symbol: RARI,
+ * //         decimals: null,
+ * //         type: 'erc-1155',
+ * //         icon: null
+ * //       }
+ * //     },
+ * //     status: 'error'
+ * //   }, ...],
+ * //   metadata: { count: 10, offset: 0, limit: 10 }
+ * // }
+ * ```
+ */
+export const getActivity = async function (
+  params: GetActivityParams
+): Promise<GetActivityResult> {
+  if (!isValidAddress(params.address)) {
+    throw new Error('Invalid address format');
+  }
+
+  // Set default pagination params
+  const {
+    address,
+    offset = DEFAULT_PAGINATION_OFFSET,
+    limit = DEFAULT_PAGINATION_LIMIT
+  } = params;
+  const baseTxRequestUrl = `https://blockscout.lisk.com/api/v2/addresses/${address}/transactions`;
+
+  const cacheKeyTransactions = address;
+  const cacheKeyNextPageParams = `${address}_params`;
+
+  const userTransactions: BlockscoutTransaction[] = activityCache.has(
+    cacheKeyTransactions
+  )
+    ? (activityCache.get(cacheKeyTransactions) as BlockscoutTransaction[])
+    : [];
+
+  let nextPageParams: BlockscoutNextPageParams | null = activityCache.has(
+    cacheKeyNextPageParams
+  )
+    ? (activityCache.get(cacheKeyNextPageParams) as BlockscoutNextPageParams)
```

Comment:
if we check for existing data and its sufficient in the case then return immediately?